### PR TITLE
Support IPinfo database configurations

### PR DESCRIPTION
**Pull request by joegallo**
Adds rest-layer support for IPinfo database configurations in the databases APIs (it's a new `NamedWriteable` implementation of the `DatabaseConfiguration.Provider` interface). They're still not actually hooked up to anything (you can't **use** them in a meaningful way), hence `>non-issue` rather than `>feature` (the `>feature` PR that this is building towards is coming very soon).

### CODE REVIEW

File: modules/ingest-geoip/src/yamlRestTest/resources/rest-api-spec/test/ingest_geoip/40_geoip_databases.yml [hunk 1]
```diff
@@ -1,8 +1,24 @@
...
+      ingest.delete_ip_location_database:
+        id: "my_database_1"
+        ignore: 404
+  - do:
+      ingest.delete_ip_location_database:
+        id: "my_database_2"
+        ignore: 404
+  - do:
+      ingest.delete_ip_location_database:
+        id: "my_database_3"
```
Comment (masseyke) [inline modules/ingest-geoip/src/yamlRestTest/resources/rest-api-spec/test/ingest_geoip/40_geoip_databases.yml:19]: This one is never created in this file.

Comment (joegallo) [reply modules/ingest-geoip/src/yamlRestTest/resources/rest-api-spec/test/ingest_geoip/40_geoip_databases.yml:19]: Good eye!

Comment (joegallo) [reply modules/ingest-geoip/src/yamlRestTest/resources/rest-api-spec/test/ingest_geoip/40_geoip_databases.yml]: Fixed in 116394a1fd572a852d517c60b658154a0b37b33a.

**Review by masseyke — COMMENTED** [review]
LGTM

**Pull request merged**